pkg/applicationserver/applicationserver.go

```diff
@@ -1117,6 +1117,9 @@ func (as *ApplicationServer) handleUplink(ctx context.Context, ids ttnpb.EndDevi
 	} else {
 		uplink.Locations = isDev.GetLocations()
 	}
+	if dev.VersionIDs != nil {
+		uplink.VersionIDs = dev.VersionIDs
```

Comment:
If this actually works, I think we have a bug somewhere, because on L1114 you don't actually request `version_ids`, so those should always be `nil`.

Reply:
The `version_ids` field is fetched from the AS device registry, not the IS.

Reply:
Ah sorry, I looked in the wrong place.